bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/AbstractRemoteAddonService.java

```diff
@@ -90,8 +91,16 @@ public void refreshSource() {
             return;
         }
         List<Addon> addons = new ArrayList<>();
-        installedAddonStorage.stream().map(e -> Objects.requireNonNull(gson.fromJson(e.getValue(), Addon.class)))
-                .forEach(addons::add);
+        try {
+            installedAddonStorage.stream().map(e -> Objects.requireNonNull(gson.fromJson(e.getValue(), Addon.class)))
+                    .forEach(addons::add);
+        } catch (JsonSyntaxException e) {
+            List.copyOf(installedAddonStorage.getKeys()).forEach(installedAddonStorage::remove);
+            logger.error(
```

Comment:
Perhaps it would be useful to also log the removed entries?

Reply:
Done